cafe/drivers/unittest/decorators.py

```diff
@@ -177,6 +178,23 @@ def decorator(cls):
                 class_name_new = "{0}_{1}".format(class_name, dataset.name)
                 new_class = type(class_name_new, (cls,), dataset.data)
                 new_class.__module__ = cls.__module__
+
+                # Find all test methods, add tags and other decorators,
+                # then set the appropriate test method on the new class
+                for member_name, member in inspect.getmembers(cls):
+                    if member_name.startswith("test_"):
+                        method_name, original_method = member_name, member
+
+                        @functools.wraps(original_method)
+                        @tags(*dataset.metadata["tags"])
+                        def new_method(*args, **kwargs):
+                            original_method(*args, **kwargs)
```

Comment:
Not totally necessary, but returning here rather than having an implicit return might be easier on readers.

Reply:
Yeah I don't think anything is worried about the return value of the test method, but it definitely makes sense. Added.